ArrayList/ArrayList.cs

```diff
@@ -1,8 +1,580 @@
 ﻿using System;
+using System.Text;
 
-namespace ArrayList
+namespace List
 {
-    public class ArrayList
+    public class ArrayList<T> where T : IComparable<T>
     {
+        public int Length
+        {
+            get
+            {
+                return _length;
+            }
+            private set
+            {
+                if (value >= 0)
+                {
+                    _length = value;
+                }
+                else if (value < 0)
+                {
+                    _length = 0;
+                }
+            }
+        }
+
+        public T this[int index]
+        {
+            get
+            {
+                if (index >= 0 && index < Length)
+                {
+                    return _array[index];
+                }
+
+                throw new IndexOutOfRangeException();
+            }
+            set
+            {
+                if (index >= 0 && index < _array.Length)
+                {
+                    _array[index] = value;
+                }
+
+                throw new IndexOutOfRangeException();
+            }
+        }
+
+        private int _length;
+        private T[] _array;
+        private int _initLength = 10;
+
+        public ArrayList()
+        {
+            Length = 0;
+            _array = new T[_initLength];
+        }
+
+        public ArrayList(T value)
+        {
+            if (value != null)
+            {
+                Length = 1;
+                _array = new T[_initLength];
+                _array[0] = value;
+            }
+            else
+            {
+                throw new ArgumentException("Values is null");
```

Comment:
We should not throw exception from constructors. You should change logic or use static factory solution.
Something like in [this article](https://riptutorial.com/design-patterns/example/26799/static-factory-method-csharp).
Signature example: `public static ArrayList Create(T value);`

Reply:
Corrected. I removed the exceptions.